include/ignition/math/detail/Sphere.hh

```diff
@@ -103,18 +103,24 @@ T Sphere<T>::VolumeBelow(const Planed &_plane) const
   // get nearest point to center on plane
   auto dist = _plane.Distance(Vector3d(0, 0, 0));
 
-  if(dist < -r)
+  if (dist < -r)
   {
     // sphere is completely below plane
     return Volume();
   }
-  else if(dist > r)
+  else if (dist > r)
   {
     // sphere is completely above plane
-    return 0;
+    return 0.0;
   }
 
-  auto h = r + dist;
+  // Vertical plane
+  if (_plane.Normal().Z() < 1e-6)
```

Comment:
Why should this be the case? I consider "below" to be defined by the normal. It is possible that the sphere is rotated when surfacing.

Reply:
Ah ok, I assumed the opposite, and even updated the docs :sweat_smile: 

https://github.com/ignitionrobotics/ign-math/blob/f398782fd650547d2d1be9693a9e7569bbe38e3a/include/ignition/math/Sphere.hh#L97

In that case, I think the implementation needs to be fixed to reflect that. See this test that I added, the normal doesn't seem to affect the result:

https://github.com/ignitionrobotics/ign-math/blob/f398782fd650547d2d1be9693a9e7569bbe38e3a/src/Sphere_TEST.cc#L141-L149

Reply:
Reading that doc update now, I think it's badly worded :grimacing: I meant to say that the normal's direction is not taken into account.

Reply:
Yep will need to fix that. I can merge these changes in and then work on a fix for it.